tm2py/core/emme.py

```diff
@@ -0,0 +1,475 @@
+        if not isinstance(matrix, EmmeMatrix):
+            matrix = self._scenario.emmebank.matrix(matrix)
+        if name is None:
+            name = self._generate_name(matrix)
+        if self._matrix_cache:
+            numpy_array = self._matrix_cache.get_data(matrix)
+        else:
+            numpy_array = matrix.get_numpy_data(self._scenario.id)
+        if matrix.type == "DESTINATION":
+            n_zones = len(numpy_array)
+            numpy_array = _np.resize(numpy_array, (1, n_zones))
+        elif matrix.type == "ORIGIN":
+            n_zones = len(numpy_array)
+            numpy_array = _np.resize(numpy_array, (n_zones, 1))
+        attrs = {"description": matrix.description}
+        self.write_array(numpy_array, name, attrs)
+
+    def write_clipped_array(
+        self,
+        numpy_array: NumpyArray,
+        name: str,
+        a_min: float,
+        a_max: float = None,
+        attrs: Dict[str, str] = None,
+    ):  # pylint: disable=R0913
+        """Write array with min and max values capped.
+
+        Args:
+            numpy_array: Numpy array
+            name: name to use for the OMX key
+            a_min: minimum value to clip array data
+            a_max: optional maximum value to clip array data
+            attrs: additional attribute key value pairs to write to OMX file
+        """
+        if a_max is not None:
+            numpy_array = numpy_array.clip(a_min, a_max)
+        else:
+            numpy_array = numpy_array.clip(a_min)
+        self.write_array(numpy_array, name, attrs)
+
+    def write_array(
+        self, numpy_array: NumpyArray, name: str, attrs: Dict[str, str] = None
+    ):
+        """Write array with name and optional attrs to OMX file.
+
+        Args:
+            numpy_array:
+            name:
+            attrs:
+        """
+        if self._mode not in ["a", "w"]:
+            raise Exception("{0}: open in read-only mode".format(self._file_path))
+        shape = numpy_array.shape
+        if len(shape) == 2:
+            chunkshape = (1, shape[0])
+        else:
+            chunkshape = None
+        attrs["source"] = "Emme"
+        numpy_array = numpy_array.astype(dtype="float64", copy=False)
+        self._omx_file.create_matrix(
+            name, obj=numpy_array, chunkshape=chunkshape, attrs=attrs
+        )
+
+    def read(self, name: str) -> NumpyArray:
+        """Read OMX data as numpy array (standard interface).
+
+        Args:
+            name: name of OMX matrix
+        """
+        return self._omx_file[name].read()
+
+    def read_hdf5(self, path: str) -> NumpyArray:
+        """Read data directly from PyTables interface.
+
+        Support for hdf5 formats that don't have full OMX compatibility.
+
+        Args:
+            path: hdf5 reference path to matrix data
+        """
+        return self._omx_file.get_node(path).read()
+
+
+def parse_num_processors(value: [str, int, float]):
```

Comment:
where does htis value "MAX-X" come from? 
why would it potentially be an int or float?